### DISCUSSION TITLE
Removing rule from rule repo

### DISCUSSION THREAD

**Discussion by RekikMohamed** [NRules/NRules#370] [Q&A] [answered]
Does NRules support rule removal from rule repo without recompiling all the rules?

**Comment by snikolayev** [comment]
No, rules are compiled to an immutable ISessionFactory. To add/remove rules, have to recompile.
What you can do though is add an agenda filter, where you would check if a rule is removed, to ignore activation. See https://nrules.net/articles/agenda-filters.html#defining-rule-filters-in-code

**Comment by RekikMohamed** [comment]
@snikolayev Can I modify the Agenda Filter after creating the session? Or Is the agenda immutable too?

**Comment by snikolayev** [comment, marked as answer]
The agenda filter is going to be your class you write - you can do what you want with it. E.g. you can inject a HashSet of rule names that are deleted/disabled, and check activations against that. At runtime you can add more rules to that HashSet

**Reply by RekikMohamed** [reply]
Thank you @snikolayev
